### PR TITLE
feat(Windows): Add windows support

### PR DESCRIPTION
**Pull request by lgeiger**
Welcome windows devs! 🎉 

Since we don't have `zmq-prebuilt` for electron on windows (yet),  this will rebuilt zmq from source. Windows devs will need Visual Studio 2013 and Python 2.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": ":notebook: the nteract notebook",
   "main": "index.js",
   "scripts": {
+    "postinstall": "node scripts/rebuild.js",
```
Comment (lgeiger) [inline package.json:7]: Is `postinstall` the right place for this?

Comment (rgbkrk) [reply package.json:7]: That's probably fine. `prepublish` is a likely candidate, but I don't know the lifecycle/order for `electron-packager`. If it's working, lets go with it for now. I'm assuming this doesn't run within the dist copy - anyone doing an install is in dev mode.

File: package.json [hunk 2]
```diff
@@ -34,6 +35,7 @@
     "dist:linux32": "npm run dist -- --platform=linux --arch=ia32",
     "dist:osx": "npm run dist -- --platform=darwin --arch=x64 --icon ./static/icon.icns --extend-info=./static/extend.plist --app-category-type=public.app-category.developer-tools --app-bundle-id=io.nteract.nteract",
     "dist:osx:signed": "npm run dist:osx -- --osx-sign",
+    "dist:win": "npm run dist -- --platform=win32 --icon ./static/icon.ico",
```
Comment (rgbkrk) [inline package.json:38]: Excellent

File: scripts/rebuild.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  var cmd = 'npm rebuild zmq-prebuilt --runtime=electron --target=' + version +
+    ' --disturl=https://atom.io/download/atom-shell --build-from-source'
+
+  exec(cmd, function(err, stdout, stderr) {
+    console.log(stdout);
+    console.log(stderr);
+    if (err) {
+      throw err;
+    }
+  });
```
Comment (rgbkrk) [inline scripts/rebuild.js:20]: While I'm a fan of this, what are we re-doing that's already done in [`electron-rebuild`](https://github.com/electron/electron-rebuild)?

**Review by rgbkrk — APPROVED** [review]

File: scripts/rebuild.js (hunk 3 above)
Comment (lgeiger) [reply scripts/rebuild.js:20]: I tried `electron-rebuild` and it didn't work out of the box. Probably it was missing a `--build-from-source` flag somewhere.

**Comment by rgbkrk** [comment]
Do you think we're ready to pull the trigger on switching to `zmq-static`?

**Comment by lgeiger** [comment]
If we switch to `zmq-static` linux and mac devs would need to install the extra [dependencies](https://github.com/nteract/zmq-static#installation).

I'm OK with it, but it would be harder for new contributors. On the other side we could use the latest electron then.

**Comment by rgbkrk** [comment]
I summon @jdetle to the conversation since I know he both wants new Electron and is closer to the feeling that new contributors face.

**Comment by lgeiger** [comment]
And one would need Python 2 for node-gyp.

That said, 👍 for moving to `zmq-static` (though that pretty much means reverting this PR 😄 )

**Comment by rgbkrk** [comment]
Since we're actually shipping builds, at the very least it means that trial users can just roll with nteract directly and not have to build from source.

My original hesitation to enforce all the build tooling setup was barriers to entry for those that are more on the frontend dev side, less astute with Jupyter.

**Comment by rgbkrk** [comment]
> though that pretty much means reverting this PR

We still have to do the rebuild though right?

**Comment by lgeiger** [comment]
> We still have to do the rebuild though right?

Yes that can be done with `electron-rebuild` then.

**Comment by jdetle** [comment]
A wild @jdetle has appeared! I have no idea what the value of zmq-static is right now, but in terms of installing additional dependencies, I think we're fine. We can always post some sort of caveat, "If you are having trouble installing dependencies for nteract, you're always welcome to ask questions in our [Slack](http://slack.nteract.in/)." (Of course I volunteer to help anyone having issues.)

**Comment by lgeiger** [comment]
@jdetle `zmq-static` is pretty much `zmq-prebuilt` without prebuilts 😄 

It would mean we could use the latest electron release at the cost of building zeromq ourself.

**Comment by rgbkrk** [comment]
Is there a reasonable way to make `zmq-prebuilt` rely on `zmq-static`?

**Comment by rgbkrk** [comment]
I'm only asking that because of the node kernel I've been hacking on - I want to keep it relying on prebuild - it makes install so painless.

**Comment by lgeiger** [comment]
@rgbkrk If we find a way of running [`build.js`](https://github.com/nteract/zmq-prebuilt/blob/master/build.js) on `preinstall` if no binaries are found it would even make `zmq-static` obsulet.

Definitely doable ~~but I don't know how~~.

**Comment by lgeiger** [comment]
@rgbkrk It looks like the `preinstall` script can be included into the gyp file.

This means we could rebuild `zmq-prebuilt`. I'll try it tomorrow and make a PR if it works.

**Comment by rgbkrk** [comment]
I need some sort of excited emoji.

**Comment by lgeiger** [comment]
It works! 🎉

I couldn't wait to try it 😉. The PR will come tomorrow.

File: scripts/rebuild.js (hunk 3 above)
Comment (lgeiger) [reply scripts/rebuild.js:20]: It looks like I ran into this issue: https://github.com/electron/electron-rebuild/issues/53

**Pull request merged**